From e10c955715a7a8febad0ea1c88557e1ba2019238 Mon Sep 17 00:00:00 2001
From: Sandeep <sandeepnair@tataelxsi.co.in>
Date: Wed, 4 Feb 2015 19:47:16 +0000
Subject: [PATCH 7/9] BPV-8919 : Adding the fixes in dibbler for fixing the
 segmentation faults Reason for change: Dibbler causing segmentation fault in
 master Test Procedure: Provision box in ipv6 mode and start dibbler client
 Risks: Low

Change-Id: Ib702e6df3b551657a37a0c9a9aa180db80d024bd
Signed-off-by: Sandeep <sandeepnair@tataelxsi.co.in>
---
 Port-linux/lowlevel-options-linux.c | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/Port-linux/lowlevel-options-linux.c b/Port-linux/lowlevel-options-linux.c
index c79fdbf..0407307 100644
--- a/Port-linux/lowlevel-options-linux.c
+++ b/Port-linux/lowlevel-options-linux.c
@@ -479,13 +479,18 @@ int prefix_add(const char* ifname, int ifindex, const char* prefixPlain, int pre
     if (!f) {
 	/* unable to open, so this file is missing, let's create it */
 	f = fopen(RADVD_FILE, "w");
-	fprintf(f, "#\n");
-	fprintf(f, "# Router Advertisement config file generated by Dibbler %s\n", DIBBLER_VERSION);
-	fprintf(f, "#\n");
-	fprintf(f, "\n");
+	if (f == NULL) {
+		sprintf(errorMsg, "Unable to create %s file. \n", RADVD_FILE);
+	}
+	else {
+		fprintf(f, "#\n");
+		fprintf(f, "# Router Advertisement config file generated by Dibbler %s\n", DIBBLER_VERSION);
+		fprintf(f, "#\n");
+		fprintf(f, "\n");
+	}
     }
     if (!f) {
-	sprintf(errorMsg, "Unable to open %s file.", RADVD_FILE);
+	sprintf(errorMsg, "Unable to open %s file.\n", RADVD_FILE);
 	return LOWLEVEL_ERROR_FILE;
     }
     fseek(f, 0, SEEK_END);
@@ -542,7 +547,15 @@ int prefix_del(const char* ifname, int ifindex, const char* prefixPlain, int pre
     rename(RADVD_FILE,RADVD_FILE".old");
 
     f = fopen(RADVD_FILE".old","r");
+    if (f == NULL) {
+	sprintf(errorMsg, "Unable to open %s file. \n", RADVD_FILE".old");
+	return LOWLEVEL_ERROR_FILE;
+    }
     f2 = fopen(RADVD_FILE,"w"); 
+    if (f2 == NULL) {
+	sprintf(errorMsg, "Unable to open %s file. \n", RADVD_FILE);
+	return LOWLEVEL_ERROR_FILE;
+    }
 
     snprintf(buf2, 511, "### %s start ###\n", ifname);
     while (fgets(buf,511,f)) {
-- 
2.3.1

